network/src/protocols/network/dummy.rs

```diff
@@ -132,13 +135,16 @@ pub fn setup_network() -> DummyNetwork {
     .into_iter()
     .collect();
 
+    let mut rng = StdRng::from_seed(TEST_SEED);
+
+    let mut listener_network_config = NetworkConfig::network_with_id(network_id.clone());
+    listener_network_config.random_with_peer_id(&mut rng, Some(listener_peer_id));
+    listener_network_config.listen_address = listener_addr;
     // Set up the listener network
     let mut network_builder = NetworkBuilder::new(
+        listener_network_config,
         runtime.handle().clone(),
-        network_id.clone(),
-        listener_peer_id,
         RoleType::Validator,
-        listener_addr,
```

Comment:
Are these the only three fields of `network_config`?  Or are there a multitude of other fields?

If the former, then this collapse makes a lot of sense.  If the latter, then my experience says we're better of peeling the outer layer of the onion away as we go down the rabbit hole;  put another way, if `network_builder` is not using 3/4 of the fields in `network_config`, then network_builder should never see the `network_config` object.

Reply:
I'm not sure what you mean by the only fields of `network_config`, since it has more fields than these. These are removed because they are duplicates between `NetworkBuilder` and `NetworkConfig`